shedlock-provider-jedis/src/main/java/net/javacrumbs/shedlock/provider/jedis/JedisLockProvider.java

```diff
@@ -0,0 +1,133 @@
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.javacrumbs.shedlock.provider.jedis;
+
+import net.javacrumbs.shedlock.core.LockConfiguration;
+import net.javacrumbs.shedlock.core.LockProvider;
+import net.javacrumbs.shedlock.core.SimpleLock;
+import net.javacrumbs.shedlock.support.LockException;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+
+/**
+ * Uses Redis's `SET resource-name anystring NX PX max-lock-ms-time` as locking mechanism.
+ *
+ * See https://redis.io/commands/set
+ *
+ */
+public class JedisLockProvider implements LockProvider {
+
+    private static final String KEY_PREFIX = "job-lock";
+    private static final String ENV_DEFAULT = "default";
+
+    private static final long AT_MOST_HOURS_DEFAULT = 1;
+
+    // Redis Flags
+    private static final String SET_IF_NOT_EXIST = "NX";
+    private static final String SET_EXPIRE_TIME_IN_MS = "PX";
+
+    private JedisPool jedisPool;
+    private String environment;
+    private Duration atMostHoursDefault;
+
+    public JedisLockProvider(JedisPool jedisPool) {
+        this(jedisPool, ENV_DEFAULT, AT_MOST_HOURS_DEFAULT);
+    }
+
+    public JedisLockProvider(JedisPool jedisPool, String environment) {
+        this(jedisPool, environment, AT_MOST_HOURS_DEFAULT);
+    }
+
+    public JedisLockProvider(JedisPool jedisPool, String environment, long atMostHoursDefault) {
+        this.jedisPool = jedisPool;
+        this.environment = environment;
+        this.atMostHoursDefault = Duration.of(atMostHoursDefault, ChronoUnit.HOURS);
+    }
+
+    @Override
+    public Optional<SimpleLock> lock(LockConfiguration lockConfiguration) {
+        Instant now = Instant.now();
+
+        // Get 'furthest out' configured expire time for lock TTL
+        long expireTime = Math.max(
+            Duration.between(now, lockConfiguration.getLockAtMostUntil()).toMillis(),
+            Duration.between(now, lockConfiguration.getLockAtLeastUntil()).toMillis());
+
+        String key = buildKey(lockConfiguration.getName(), this.environment);
+
+        try (Jedis jedis = jedisPool.getResource()) {
+            String rez = jedis.set(key,
+                    buildValue(),
+                    SET_IF_NOT_EXIST,
+                    SET_EXPIRE_TIME_IN_MS,
+                    expireTime < 0 ? atMostHoursDefault.toMillis() : expireTime); // And if not set, use default
```

Comment:
Yea - it was in the Spring code but wasn't making it here.  Makes sense to drop this and use a common one.